ormar/relations/relation.py

```diff
@@ -155,6 +155,9 @@ def add(self, child: "Model") -> None:
         if self._type in (RelationType.PRIMARY, RelationType.THROUGH):
             self.related_models = child
             self._owner.__dict__[relation_name] = child
+        elif self._type == RelationType.REVERSE:
+            # TODO: Define what to do with reverse relations
+            pass
```

Comment:
The reverse relation logic is include in else now. The change in pydantic v2 is that instead of calling a method to dump to json/dict it's now calling rust code which gets the whole object and basically grabs the `__dict__` directly, that's the check in `__getattribute__` but that happens also in other cases. Before ormar used to keep related models on relation/relation proxy and dump them to dict only on dict/json calls. Now I had to add the related models to __dict__ too to ensure proper dumping in pydantic. That seems to work well except for the cases in which we have circular references. As there is no exact point to remove them before dumping (before there was a relationship map in ormar that was built on dump call that excluded circular references). 

The whole problem is how to maintain access to nested models and keep the relations accessible in code, but exclude them in dumping. So, when you have models `A->B` you can still get the `B` and dump it in a way that it has the `A` in related models without the additional db call, but at the same time prevent dumping `A->B->A`.

Reply:
Pydantic offers a solution for serializing recursive fields but only on a field by field case if i understand correctly 

https://github.com/pydantic/pydantic/blob/b2aa36a80904773d7db2e4c007fc00a5a4d7653b/docs/concepts/postponed_annotations.md?plain=1#L251

If we could make this kind of "Skip if pydantic thinks its repeating" logic for all model fields by default we could escape this but it doesnt seem like a nice solution

Reply:
Nice catch! didn't see it before :) 

IIUC it would have to be only the reverse relation field and not all of them. It might be kind of ugly but hey, there are already some ugly workarounds in ormar anyway 🤣 I want to use the v2 to do some refactoring if possible, but the primary goal is to make it work with v2.